erts/emulator/beam/jit/x86/beam_asm.hpp

```diff
@@ -795,7 +795,20 @@ class BeamAssembler : public ErrorHandler {
     void mov_imm(x86::Gp to, T value) {
         static_assert(std::is_integral<T>::value || std::is_pointer<T>::value);
         if (value) {
-            a.mov(to, imm(value));
+            /* Generate the shortest instruction to set the register to an
```

Comment:
This should already have been handled by `asmjit` since we pass `kEncodingOptionOptimizeForSize`, perhaps we should report this as a bug instead?

Reply:
I suggest we report this issue and keep this change until we have a fixed and updated `asmjit` in place. (We still need to keep the `mov_imm()` helper since it will use `xor` to zero the register, something that `asmjit` is not allowed to do.)